src/annotator/guest.ts

```diff
@@ -517,12 +535,23 @@ export class Guest extends TinyEmitter implements Annotator, Destroyable {
 
     this._hostRPC.on('close', () => this.emit('hostDisconnected'));
 
+    // Schedule messages to be sent after connection to host is established.
+    this._notifySupportedToolsChanged();
```

Comment:
Perhaps the comment is a bit misleading, since the fact that messages are scheduled until the connection is ready, is an implementation detail in `PortRPC`, but not immediately obvious by just looking at `_notifySupportedToolsChanged` implementation.

Reply:
This behavior of `PortRPC.call` is part of its contract, ie. it is promised in the documentation. More generally it just has to work this way because the interface would become really awkward to use otherwise. All code that invokes `call` would have to handle the possibility that the connection is not established yet and do its own queueing.